src/sentry/models/user.py

```diff
@@ -381,12 +387,45 @@ def _normalize_before_relocation_import(
             self.is_staff = False
             self.is_superuser = False
 
-        # TODO(getsentry/team-ospo#181): Handle usernames that already exist. This will involve
-        # marking the user "unclaimed", wiping their password, and adding a random suffix to their
-        # username.
+        lock = locks.get(f"user:username:{self.id}", duration=5, name="username")
```

Comment:
How was the duration decided? Are we sure that's enough time?

Reply:
The other locks tend to use this number most often. I figured we could start here, then adjust as needed.